distribution/docs/src/main/jdocs/content/_security-framework/realms-contents.adoc

```diff
@@ -0,0 +1,107 @@
+
+|org.codice.${ddf-branding-lowercase}.security.filter.login.LoginFilter
+|This filter runs immediately after the WebSSOFilter and exchanges any authentication information found in the request with a Subject via Shiro.
+
+|org.codice.${ddf-branding-lowercase}.security.filter.authorization.AuthorizationFilter
+|This filter runs immediately after the `LoginFilter` and checks any permissions assigned to the web context against the attributes of the user via Shiro.
+
+|org.apache.shiro.realm.AuthenticatingRealm
+|This is an abstract authenticating realm that exchanges an `org.apache.shiro.authc.AuthenticationToken` for a `${ddf-branding-lowercase}.security.Subject` in the form of an `org.apache.shiro.authc.AuthenticationInfo`
+
+|${ddf-branding-lowercase}.security.realm.sts.StsRealm
+|This realm is an implementation of `org.apache.shiro.realm.AuthenticatingRealm` and connects to an STS (configurable) to exchange the authentication token for a Subject.
+
+|${ddf-branding-lowercase}.security.service.AbstractAuthorizingRealm
+|This is an abstract authorizing realm that takes care of caching and parsing the Subject's `AuthorizingInfo` and should be extended to allow the implementing realm to focus on making the decision.
+
+|${ddf-branding-lowercase}.security.pdp.realm.AuthZRealm
+|This realm performs the authorization decision and may or may not delegate out to the external XACML processing engine. It uses the incoming permissions to create a decision.
+However, it is possible to extend this realm using the ${ddf-branding-lowercase}.security.policy.extension.PolicyExtension interface.
+This interface allows an integrator to add additional policy information to the PDP that can't be covered via its generic matching policies.
+This approach is often easier to configure for those that are not familiar with XACML.
+Note that no `PolicyExtension` implementations are provided out of the box.
+
+|org.codice.${ddf-branding-lowercase}.security.validator.*
+|A number of STS validators are provided for X.509 (BinarySecurityToken), UsernameToken, SAML Assertion, and ${branding} custom tokens.
+The ${branding} custom tokens are all `BinarySecurityTokens` that may have PKI or username/password information as well as an authentication realm (correlates to JAAS realms installed in the container).
+The authentication realm allows an administrator to restrict which services they wish to use to authenticate users.
+For example: installing the `security-sts-ldaplogin` feature will enable a JAAS realm with the name "ldap".
+This realm can then be specified on any context using the Web Context Policy Manager.
+That realm selection is then passed via the token sent to the STS to determine which validator to use.
+
+|===
+
+[WARNING]
+====
+An update was made to the SAML Assertion Handler to pass SAML assertions via headers instead of cookies.
+Cookies are still accepted and processed to maintain legacy federation compatibility, but only headers are used when federating out.
```

Comment:
❗️ "federating out" sounds folksy, or at least idiomatic. I'm not sure what the correct substitution is. Maybe: "but only HTTP headers are used pass SAML assertions when federating with other ${branding} servers"

Reply:
I like that. I agree on the `out` being too informal.